.github/workflows/vulnerability-scan.yaml

```
@@ -0,0 +1,80 @@
+      with:
+        run_install: false
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '22'
+        cache: 'pnpm'
+    - name: Install dependencies
+      run: pnpm install --frozen-lockfile
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
```

actionlint (1.7.4)

the runner of "github/codeql-action/analyze@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)